tests/components/persistent_notification/test_init.py

```diff
@@ -175,3 +178,104 @@ async def test_manual_notification_id_round_trip(hass: HomeAssistant) -> None:
     )
 
     assert len(notifications) == 0
+
+
+async def test_automation_with_pn_trigger(hass: HomeAssistant) -> None:
+    """Test automation with a persistent_notification trigger."""
+
+    result_any = []
+    result_dismissed = []
+    result_id = []
+
+    trigger_info = {"trigger_data": {}}
+
+    @callback
+    def trigger_callback_any(
+        run_variables: dict[str, Any], context: Context | None = None
+    ) -> None:
+        result_any.append(run_variables)
+
+    await trigger.async_attach_trigger(
+        hass,
+        {"platform": "persistent_notification"},
+        trigger_callback_any,
+        trigger_info,
+    )
+
+    @callback
+    def trigger_callback_dismissed(
+        run_variables: dict[str, Any], context: Context | None = None
+    ) -> None:
+        result_dismissed.append(run_variables)
+
+    await trigger.async_attach_trigger(
+        hass,
+        {"platform": "persistent_notification", "update_type": "removed"},
+        trigger_callback_dismissed,
+        trigger_info,
+    )
+
+    @callback
+    def trigger_callback_id(
+        run_variables: dict[str, Any], context: Context | None = None
+    ) -> None:
+        result_id.append(run_variables)
+
+    await trigger.async_attach_trigger(
+        hass,
+        {"platform": "persistent_notification", "notification_id": "42"},
+        trigger_callback_id,
+        trigger_info,
+    )
+
+    await hass.services.async_call(
+        pn.DOMAIN,
+        "create",
+        {"notification_id": "test_notification", "message": "test"},
+        blocking=True,
+    )
+
+    await hass.async_block_till_done()
+
+    result = result_any[0].get("trigger")
+    assert result.get("platform") == "persistent_notification"
+    assert result.get("update_type") == pn.UpdateType.ADDED
+    assert result.get("notification", {}).get("notification_id") == "test_notification"
+    assert result.get("notification", {}).get("message") == "test"
+
+    assert len(result_dismissed) == 0
+    assert len(result_id) == 0
+
+    await hass.services.async_call(
+        pn.DOMAIN,
+        "dismiss",
+        {"notification_id": "test_notification"},
+        blocking=True,
+    )
+
+    await hass.async_block_till_done()
+
+    result = result_any[1].get("trigger")
+    assert result.get("platform") == "persistent_notification"
+    assert result.get("update_type") == pn.UpdateType.REMOVED
+    assert result.get("notification", {}).get("notification_id") == "test_notification"
+    assert result.get("notification", {}).get("message") == "test"
```

Comment:
```suggestion
    assert result["platform"] == "persistent_notification"
    assert result["update_type"] == pn.UpdateType.REMOVED
    assert result["notification"]["notification_id"] == "test_notification"
    assert result["notification"]["message"] == "test"
```

Use `[]` when we expect the value to be present